rust/lance-index/src/vector/hnsw.rs

```diff
@@ -305,24 +334,30 @@ impl HNSW {
             .collect())
     }
 
-    /// Write the HNSW graph to a Lance file.
-    pub async fn write(&self, writer: &mut FileWriter<ManifestDescribing>) -> Result<()> {
-        let mut level_offsets = vec![0];
+    /// Returns the metadata of this [`HNSW`].
+    pub fn metadata(&self) -> HnswMetadata {
+        let mut level_offsets = Vec::with_capacity(self.levels.len());
+        let mut offset = 0;
         for level in self.levels.iter() {
-            level_offsets.push(level_offsets.last().unwrap() + level.len());
-            // TODO: add chunking to each batch.
-            writer.write(&[level.nodes.clone()]).await?;
+            level_offsets.push(offset);
+            offset += level.len();
```

Comment:
do we need one more offset in the level_offsets so that it is easie to find the right side of the range. 

For example, if each levels has [10, 20] nodes, the offset can be `[0, 10, 30]`.